Assets/Scripts/Fire.cs

```diff
@@ -0,0 +1,38 @@
+﻿using UnityEngine;
+
+public class Fire : MonoBehaviour {
+    public float scaleFire;
+    public float speedDreasing;
+    public float minScale;
+    public int densityFire;
+    private ParticleSystem part;
+    private ParticleSystem.ShapeModule sh;
+    private ParticleSystem.EmissionModule em;
+
+    void Start() {
+        part = this.GetComponent<ParticleSystem>();
+        sh = part.shape;
+        em = part.emission;
+        sh.scale = Vector3.ClampMagnitude(Vector3.one, scaleFire);
+        em.rateOverTime = (ParticleSystem.MinMaxCurve)(System.Math.Pow(scaleFire, 3) * densityFire);
+    }
+
+    void Update() {
+        Collider[] woods = Physics.OverlapSphere(this.transform.position, sh.scale.x + 1);
+        foreach (var wood in woods) {
+            if (!part.isStopped && wood.gameObject.GetComponent<Wood>()) {
```

Comment:
Instead of a foreach with an if you can filter the elements out (or at least invert if to reduce nesting, Rider has function for this as well)
Alternatively you can save the results of GetComponent - it is an expensive operation and you're calling it again the line below.